modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteSnapshotManager.java

```diff
@@ -1710,75 +1725,23 @@ public IgniteInternalFuture<SnapshotPartitionsVerifyTaskResult> checkSnapshot(
-            (node) -> CU.baselineNode(node, kctx0.state().clusterState()));
-
-        Collection<Integer> grpIds = grps == null ? Collections.emptySet() : F.viewReadOnly(grps, CU::cacheId);
-
-        SnapshotMetadataVerificationTaskArg taskArg = new SnapshotMetadataVerificationTaskArg(name, snpPath, incIdx, grpIds);
-
-        kctx0.task().execute(
-            SnapshotMetadataVerificationTask.class,
-            taskArg,
-            options(bltNodes)
-        ).listen(f0 -> {
-            SnapshotMetadataVerificationTaskResult metasRes = f0.result();
+        IgniteInternalFuture<SnapshotPartitionsVerifyTaskResult> res = checkSnpProc.start(
+            name, snpPath, grps, check, incIdx, includeCustomHandlers);
 
-            if (f0.error() == null && F.isEmpty(metasRes.exceptions())) {
-                Map<ClusterNode, List<SnapshotMetadata>> metas = metasRes.meta();
-
-                Class<? extends AbstractSnapshotVerificationTask> cls;
-
-                if (includeCustomHandlers)
-                    cls = SnapshotHandlerRestoreTask.class;
-                else
-                    cls = incIdx > 0 ? IncrementalSnapshotVerificationTask.class : SnapshotPartitionsVerifyTask.class;
-
-                kctx0.task().execute(
-                        cls,
-                        new SnapshotPartitionsVerifyTaskArg(grps, metas, snpPath, incIdx, check),
-                        options(new ArrayList<>(metas.keySet()))
-                    ).listen(f1 -> {
-                        if (f1.error() == null)
-                            res.onDone(f1.result());
-                        else if (f1.error() instanceof IgniteSnapshotVerifyException) {
-                            IdleVerifyResult idleRes = IdleVerifyResult.builder()
-                                .exceptions(((IgniteSnapshotVerifyException)f1.error()).exceptions()).build();
-
-                            res.onDone(new SnapshotPartitionsVerifyTaskResult(metas, idleRes));
-                        }
-                        else
-                            res.onDone(f1.error());
-                    });
-            }
-            else {
-                if (f0.error() == null)
-                    res.onDone(new IgniteSnapshotVerifyException(metasRes.exceptions()));
-                else if (f0.error() instanceof IgniteSnapshotVerifyException) {
-                    IdleVerifyResult idleRes = IdleVerifyResult.builder()
-                        .exceptions(((IgniteSnapshotVerifyException)f0.error()).exceptions()).build();
+        res.listen(lsnr -> {
+            if (log.isInfoEnabled()) {
+                Throwable err = lsnr.error();
 
-                    res.onDone(new SnapshotPartitionsVerifyTaskResult(null, idleRes));
-                }
-                else
-                    res.onDone(f0.error());
+                log.info("The check snapshot procedure finished [success=" + (err == null) + ", snpName=" + name + ", snpPath=" + snpPath
```

Comment:
With
`success=" + (err == null) +`
we duplicate 
`(err == null ? "" : ", err=" + err.getMessage()) `. 

Also `snpName=` is essentially expected first.